src/libsync/propagateuploadencrypted.h

```diff
@@ -36,7 +36,7 @@ class PropagateUploadEncrypted : public QObject
     void start();
 
     /* unlocks the current folder that holds this file */
-    void unlockFolder();
+    void unlockFolder(bool uploadFailed = false);
```

Comment:
enum please ;-)

or... we got a couple of alternatives here I think:
 * receive a std::function which would be executed before `_isUnlockRunning` is set back to false, so instead of passing `true` you'd explicitly pass a lambda which does the emit
 * connect `folderUnlocked` to `error` before calling `unlockFolder` in the cases you're currently passing `true`